src/main/java/command/Command.java

```diff
@@ -0,0 +1,33 @@
+package command;
+
+
+import duke.DukeException;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+
+/**
+ * Abstract command which has many different type of commands.
+ *
+ */
+public abstract class Command {
+    private boolean isExit;
+
+
+    /**
+     * Executes the given line instruction passed into the object.
+     *
+     * @param tasklist Tasklist object containing the data structure.
+     * @param ui duke Ui object responsible for interface response.
+     * @param storage duke Storage object to make changes to hard drive data.
+     */
+    public abstract void execute(TaskList tasklist, Ui ui, Storage storage) throws DukeException;
```

Comment:
It may be worth considering changing the `execute` method of `Command` and its child classes such that executing the command returns a result of some type, which the main `Duke` class would pass to its `Ui` instance to handle. This would de-couple commands from the class that handles UI and allow it to be more flexible.